src/pages/Timetable.tsx

```
@@ -19,13 +22,62 @@ const TimetablePage: React.FC = () => {
+
+  // Use appropriate hook based on view mode
+  const listResult = useTimetableList(viewMode === 'List' ? apiParams : undefined);
+  const cardResult = useTimetableCard(viewMode === 'Card' ? apiParams : undefined);
+
+  // Select the active result based on view mode
+  const activeResult = viewMode === 'List' ? listResult : cardResult;
+  const { data, totalPages, loading, error, refetch } = activeResult;
+
+  // Update API params when filters or search queries change
+  useEffect(() => {
+    const params: any = {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)